### PR TITLE
feat: Drag and drop

### PR DESCRIPTION
**Pull request by mbssantos**
Drag and drop functionality.
Allows user to reorder tiles and move them across windows without tearouts.
Annuls #770 and #534

### CODE REVIEW

**Comment by jdhodges** [comment]
Really nice work, just something I noticed. The reorder tiles feature doesn't seem to be working correctly.

Here I'm taking the top tile and dragging it to the position under TSLA:
image

Dropping it here results in:
image

Two things to point out:
1) MSFT should now be in the second position.
2) There should only be one 'empty' space during the dnd operation (the destination of the dragged tile). The other tiles should transition up to fill the space left by the dragged tile.

**Comment by mbssantos** [comment]
Addressed those two issues.
For future reference, this already inlcudes cross window drag & drop

**Comment by jleft** [comment]
:+1:

File: src/child/containers/sidebars/favourites/Favourites.js [hunk 1]
```diff
@@ -38,14 +58,56 @@ class Favourites extends Component {
         this.props.dispatch(selectStock(stockCode, stockName));
     }
 
-    onDrag() {
-        // TODO: send content to tearout
+    onDragOverFavourite(e, targetCode) {
+        const codes = this.props.favourites.codes;
+        const code = getCodeFromDT(e.dataTransfer.types);
+        // const index = codes.indexOf(targetCode);
```
Comment (jdhodges) [inline src/child/containers/sidebars/favourites/Favourites.js]: Can we remove this?

**Comment by rjmcneill** [comment]
:+1:

File: src/child/components/Favourite.js [hunk 2]
```diff
@@ -18,6 +17,7 @@ class Favourite extends Component {
 
     componentDidMount() {
         const stockCode = this.props.stockCode;
+        this.addDragTarget(stockCode);
```
Comment (mic-smith) [inline src/child/components/Favourite.js]: Should remove the event listeners created in this call in componentWillUnmount

**Comment by jleft** [comment]
Merged with the intention that the 2 issues listed above will be addressed separately (removing redundant comment (https://github.com/ScottLogic/StockFlux/pull/809#discussion_r64353516) and removing event listeners (https://github.com/ScottLogic/StockFlux/pull/809#discussion_r64578393))

**Pull request merged**